File: c_Programming/Assingements/Assignment 14/pass by address/salemanager.c
#include<stdio.h>
typedef struct SalesManager
{
	int id;
	char name[20];
	double salary;
	double incentive;
	int target;
	
}SalesManager;
void storesalemanagerdata(SalesManager*);
void display(SalesManager*);
void main()
{
	SalesManager sm1,sm2;
	storesalemanagerdata(&sm1);
	display(&sm1);
	
	storesalemanagerdata(&sm2);
	display(&sm2);
	
}
void storesalemanagerdata(SalesManager* salemang)
{
	printf("\nEnter the SalesManager 1 Id , name and salary , incentive & target:");
	scanf("%d%s%lf%lf%d",&salemang->id,salemang->name,&salemang->salary,&salemang->incentive,&salemang->target);
}
void display(SalesManager* salemang)
{
	printf("\nSalesManagerId :%d \nSalesManager name :%s \nsalary :%.2lf  \nincentive :%.2lf  \ntarget :%d",salemang->id,salemang->name,salemang->salary,salemang->incentive,salemang->target);
	
}

File: Java_Module/demo java/GetterSetter/SaleManager.java
class SalesManager {
    int id;
    String name;
    double salary;
    double incentive;
    int target;

    // SET methods
    void setId(int i) {
        this.id = i;
    }

    void setName(String n) {
        this.name = n;
    }

    void setSalary(double s) {
        this.salary = s;
    }

    void setIncentive(double i) {
        this.incentive = i;
    }

    void setTarget(int t) {
        this.target = t;
    }

    // GET methods
    int getId() {
        return this.id;
    }

    String getName() {
        return this.name;
    }

    double getSalary() {
        return this.salary;
    }

    double getIncentive() {
        return this.incentive;
    }

    int getTarget() {
        return this.target;
    }

    // DISPLAY method using GET methods
    void display() {
        System.out.println("SalesManager Id : " + getId());
        System.out.println("SalesManager Name : " + getName());
        System.out.println("SalesManager Salary : " + getSalary());
        System.out.println("SalesManager Incentive : " + getIncentive());
        System.out.println("SalesManager Target : " + getTarget());
    }
}

class SalesManagerData {
    public static void main(String args[]) {

        System.out.println("SalesManager Details");

        SalesManager sm1, sm2;

        sm1 = new SalesManager();
        sm2 = new SalesManager();

        // First SalesManager details using SET
        sm1.setId(101);
        sm1.setName("Divya");
        sm1.setSalary(40000);
        sm1.setIncentive(5000);
        sm1.setTarget(100);

        // Second SalesManager details using SET
        sm2.setId(102);
        sm2.setName("Pratu");
        sm2.setSalary(45000);
        sm2.setIncentive(6000);
        sm2.setTarget(120);

        // Display
        sm1.display();
        sm2.display();
    }
}